lib/collection.js

```diff
@@ -382,12 +383,7 @@ Collection.prototype.find = function(query, options, callback) {
   // Translate to new command option noCursorTimeout
   if (typeof newOptions.timeout === 'boolean') newOptions.noCursorTimeout = newOptions.timeout;
 
-  // Merge in options to command
-  for (let name in newOptions) {
-    if (newOptions[name] != null && name !== 'session') {
-      findCommand[name] = newOptions[name];
-    }
-  }
+  decorateCommand(findCommand, newOptions, { session: true, collation: true });
```

Comment:
This is a little out of scope for this PR, but maybe worth it since `decorateCommand` only appears to be used one other place in the codebase. Can we do better here than providing an object for exclusion? I'm thinking just an array `decorateCommand(findCommand, newOptions, [ 'session', 'collation' ]);`, or a more verbose ` decorateCommand(findCommand, newOptions, { exclude: [ 'session', 'collation' ]);` if that feels more self-documenting.

Reply:
is it worth using a Set rather than an Array for the faster checking time? i.e. `if (!exclude.has(name)) command[name] = options[name]` rather than iterating over the array with `array.includes()`

Reply:
Personally I think the ergonomics of array are better here, and also as this [jsperf](https://jsperf.com/has-vs-indexof-10/1) anecdotally indicates for small enough sets Array.indexOf is actually faster than set lookup.